File: src/main/java/wtf/nucker/pluginname/PluginName.java
package wtf.nucker.pluginname;

import de.leonhard.storage.Yaml;
import org.bukkit.plugin.java.JavaPlugin;

public class PluginName extends JavaPlugin  {

    private Yaml config;

    @Override
    public void onEnable() {
        this.config = new Yaml("config.yml", this.getDataFolder().getAbsolutePath(), this.getResource("config.yml"));
    }

    public Yaml getConfigFile() {
        return config;
    }
}

File: src/main/kotlin/wtf/nucker/pluginName/PluginName.kt
package wtf.nucker.pluginName

import de.leonhard.storage.Yaml
import de.leonhard.storage.internal.settings.ConfigSettings
import org.bukkit.plugin.java.JavaPlugin

class PluginName: JavaPlugin() {

    private lateinit var config: Yaml
    private lateinit var commandManager: CommandManager

    override fun onEnable() {
        this.config = Yaml("config.yml", this.dataFolder.absolutePath, this.getResource("config.yml"),
            null, ConfigSettings.PRESERVE_COMMENTS, null).addDefaultsFromInputStream()
        this.commandManager = CommandManager(this)
    }
}